tests/multigrid/constrained_dofs_06.cc

```diff
@@ -0,0 +1,75 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/exceptions.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/multigrid/mg_constrained_dofs.h>
+
+#include <vector>
+
+#include "../tests.h"
+
+
+template <int dim>
+void
+test()
+{
+  Triangulation<dim> triangulation(
+    Triangulation<dim>::limit_level_difference_at_vertices);
+  GridGenerator::hyper_cube(triangulation, 0., 1., true);
+  triangulation.refine_global(4 - dim);
+
+  DoFHandler<dim> dof_handler(triangulation);
+  dof_handler.distribute_dofs(FESystem<dim>(FE_Q<dim>(1), dim));
+  dof_handler.distribute_mg_dofs();
+
+  std::set<types::boundary_id> no_flux_boundary;
```

Comment:
maybe just use ``= {0,1,2};`` in 2d and 3d? Shorter code and you test not setting all boundaries.

Reply:
I think you can simplify the initialization here if you follow @tjhei 's suggestion